subsys/bluetooth/audio/bap_broadcast_assistant.c

```diff
@@ -594,6 +595,63 @@ static struct bt_le_scan_cb scan_cb = {
 
 /****************************** PUBLIC API ******************************/
 
+static int broadcast_assistant_reset(struct bap_broadcast_assistant_instance *inst)
+{
+	inst->busy = false;
+	inst->scanning = false;
+	inst->pa_sync = 0U;
+	inst->recv_state_cnt = 0U;
+	inst->start_handle = 0U;
+	inst->end_handle = 0U;
+	inst->cp_handle = 0U;
+
+	for (int i = 0U; i < CONFIG_BT_BAP_BROADCAST_ASSISTANT_RECV_STATE_COUNT; i++) {
+		inst->src_ids[i] = 0U;
+		inst->past_avail[i] = false;
+		inst->recv_state_handles[i] = 0U;
+		inst->src_ids[i] = 0U;
+	}
+
+	if (inst->conn != NULL) {
+		struct bt_conn *conn = inst->conn;
+		struct bt_conn_info info;
+		int err;
+
+		err = bt_conn_get_info(conn, &info);
+		if (err != 0) {
+			return err;
+		}
+
+		if (info.state == BT_CONN_STATE_CONNECTED) {
+			for (size_t i = 0U; i < ARRAY_SIZE(inst->recv_state_sub_params); i++) {
+				/* It's okay if this fail with -EINVAL as that means that they are
+				 * not currently subscribed
+				 */
+				err = bt_gatt_unsubscribe(conn, &inst->recv_state_sub_params[i]);
+				if (err != 0 && err != -EINVAL) {
+					LOG_DBG("Failed to unsubscribe to state: %d", err);
+
+					return err;
+				}
+			}
+		}
+
+		bt_conn_unref(conn);
+		inst->conn = NULL;
+	}
+
+	return 0;
```

Comment:
There are some members of inst that we dont seem to reset, do we not need to do that?
recv_state_disc_params
read_params
write_params
disc_params